app/src/main/java/com/example/photostomusic/fragments/CameraFragment.java

```diff
@@ -161,14 +163,14 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
         // Get the name of hex of each color resource as a string to fill the map
         // The colors are recovered as an int, then logical AND operation to remove
         // the transparency of the color, convert this to a HEX string
-        String color1 = Integer.toHexString(getResources().getColor(R.color.option1) & 0x00ffffff);
-        String color2 = Integer.toHexString(getResources().getColor(R.color.option2) & 0x00ffffff);
-        String color3 = Integer.toHexString(getResources().getColor(R.color.option3) & 0x00ffffff);
-        String color4 = Integer.toHexString(getResources().getColor(R.color.option4) & 0x00ffffff);
-        String color5 = Integer.toHexString(getResources().getColor(R.color.option5) & 0x00ffffff);
-        String color6 = Integer.toHexString(getResources().getColor(R.color.option6) & 0x00ffffff);
-        String color7 = Integer.toHexString(getResources().getColor(R.color.option7) & 0x00ffffff);
-        String color8 = Integer.toHexString(getResources().getColor(R.color.option8) & 0x00ffffff);
+        String color1 = Integer.toHexString(getResources().getColor(R.color.option1) & HEX_BASE);
+        String color2 = Integer.toHexString(getResources().getColor(R.color.option2) & HEX_BASE);
+        String color3 = Integer.toHexString(getResources().getColor(R.color.option3) & HEX_BASE);
+        String color4 = Integer.toHexString(getResources().getColor(R.color.option4) & HEX_BASE);
+        String color5 = Integer.toHexString(getResources().getColor(R.color.option5) & HEX_BASE);
+        String color6 = Integer.toHexString(getResources().getColor(R.color.option6) & HEX_BASE);
+        String color7 = Integer.toHexString(getResources().getColor(R.color.option7) & HEX_BASE);
+        String color8 = Integer.toHexString(getResources().getColor(R.color.option8) & HEX_BASE);
```

Comment:
same here, lots of repeated code that could be pulled out. in fact, the HEX_BASE and this Int.toHex stuff could all be pulled out into more reusable code

Reply:
Done :)